Spec/RealtimeClientConnection.swift

```diff
@@ -3028,26 +2989,165 @@ class RealtimeClientConnection: QuickSpec {
                     default:
                         fail("unhandled: \(fixtureMessage["expectedType"].string!)")
                     }
+                }
 
-                    waitUntil(timeout: testTimeout) { done in
-                        channel.publish([message]) { err in
-                            if let err = err {
-                                fail("\(err)")
+                // https://github.com/ably/wiki/issues/22
+                it("should encode and decode fixture messages as expected") {
+                    let options = AblyTests.commonAppSetup()
```

Comment:
> at what point do we publish with a raw JSON message

[Here](https://github.com/ably/ably-ios/pull/459/files/3176c628c244243510171e5852d62780c41df921..1ee851cb4020e66c1a893573bd687aca9076f0a2#diff-8c3eccfb7635c28b9fadd0e4fcb6e6b0R3019). That's a `POST /channels/.../messages`. `fixtureMessage` is a `JSON` object (from a library for easy JSON manipulation) and `rawData()` converts it into a JSON string. That's quite like cutting and pasting from the `messages-encoding.json` directly into the POST request body.

> and get that message over a JSON transport

[Here](https://github.com/ably/ably-ios/pull/459/files/3176c628c244243510171e5852d62780c41df921..1ee851cb4020e66c1a893573bd687aca9076f0a2#diff-8c3eccfb7635c28b9fadd0e4fcb6e6b0R3054). `data` there is the HTTP response body, which the `JSON` library parses.

> and confirm the message.data is as expected

[Here](https://github.com/ably/ably-ios/pull/459/files/3176c628c244243510171e5852d62780c41df921..1ee851cb4020e66c1a893573bd687aca9076f0a2#diff-8c3eccfb7635c28b9fadd0e4fcb6e6b0R3055). Both `persistedMessage` and `fixtureMessage` are `JSON` objects that don't go through the `ARTRealtime` instance at any point. The `ARTRealtime` instance is used to test that the iOS library encodes and decodes the data as expected (ie. as `expectedValue` or `expectedHexValue` tell us), be it JSON, string or base64.

> As above, and where do we publish over JSON using the library, and then confirm with a raw HTTP request that the JSON in Ably is as expected?

That's in the second test, [here](https://github.com/ably/ably-ios/pull/459/files/3176c628c244243510171e5852d62780c41df921..1ee851cb4020e66c1a893573bd687aca9076f0a2#diff-8c3eccfb7635c28b9fadd0e4fcb6e6b0R3069).